From 8c719e469054c4c74bc4152c7c6619f87f9c7035 Mon Sep 17 00:00:00 2001
From: Aswin Murugan <quic_aswinm@quicinc.com>
Date: Fri, 2 May 2025 20:37:33 +0530
Subject: [PATCH] PENDING: Remove unwanted DTS entries and configs in qcs9100

Removed unwanted device tree entries and Configs to
optimize boot time.

Signed-off-by: Aswin Murugan <quic_aswinm@quicinc.com>
Upstream-Status: Pending
---
 arch/arm/dts/qcs9100-ride-r3-u-boot.dtsi | 255 +++++++++++++++++++++++
 configs/qcs9100_defconfig                |  70 +++++++
 2 files changed, 325 insertions(+)

diff --git a/arch/arm/dts/qcs9100-ride-r3-u-boot.dtsi b/arch/arm/dts/qcs9100-ride-r3-u-boot.dtsi
index 20f556f1fe2..7da40bdc99d 100644
--- a/arch/arm/dts/qcs9100-ride-r3-u-boot.dtsi
+++ b/arch/arm/dts/qcs9100-ride-r3-u-boot.dtsi
@@ -110,4 +110,259 @@
 		};
 	};
 };
+/ {
+	aliases {
+		/delete-property/ ethernet0;
+		/delete-property/ ethernet1;
+		/delete-property/ serial1;
+		/delete-property/ serial2;
+		/delete-property/ spi16;
+	};
+	soc@0 {
+		clock-controller@100000 {
+			clocks = <&rpmhcc RPMH_CXO_CLK>,
+				<&sleep_clk>,
+				<0>,
+				<0>,
+				<0>,
+				<&usb_0_qmpphy>,
+				<&usb_1_qmpphy>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>,
+				<0>;
+		};
+		/delete-node/ funnel@4041000;
+		/delete-node/ funnel@4045000;
+		/delete-node/ funnel@4b04000;
+		/delete-node/ funnel@4042000;
+		/delete-node/ funnel@6810000;
+		/delete-node/ funnel@6800000;
+		/delete-node/ tpdm@4003000;
+		/delete-node/ tpdm@400f000;
+		/delete-node/ tpdm@4b09000;
+		/delete-node/ tpdm@4b0a000;
+		/delete-node/ tpdm@4b0b000;
+		/delete-node/ tpdm@4b0c000;
+		/delete-node/ tpdm@4b0d000;
+		/delete-node/ tpdm@6860000;
+		/delete-node/ tpdm@6861000;
+		/delete-node/ tpdm@68a0000;
+		/delete-node/ tpdm@68b0000;
+		/delete-node/ tpdm@68c0000;
+		/delete-node/ tpda@4004000;
+		/delete-node/ tpda@4b08000;
+		/delete-node/ tpda@6863000;
+		/delete-node/ etm@6040000;
+		/delete-node/ etm@6140000;
+		/delete-node/ etm@6240000;
+		/delete-node/ etm@6340000;
+		/delete-node/ etm@6440000;
+		/delete-node/ etm@6540000;
+		/delete-node/ etm@6640000;
+		/delete-node/ etm@6740000;
+		/delete-node/ replicator@4b06000;
+		/delete-node/ sram@c3f0000;
+		usb@a6f8800 {
+			/delete-property/ interrupts-extended;
+		};
+		usb@a8f8800 {
+			/delete-property/ interrupts-extended;
+		};
+		usb@a4f8800 {
+			/delete-property/ interrupts-extended;
+		};
+		spmi@c440000 {
+			/delete-property/ interrupts-extended;
+		};
+		pinctrl@f000000 {
+			/delete-property/ wakeup-parent;
+			/delete-node/ ethernet0-default-state;
+			/delete-node/ pcie0-default-state;
+			/delete-node/ pcie1-default-state;
+			/delete-node/ qup-spi16-state;
+			/delete-node/ qup-uart12-state;
+			/delete-node/ qup-uart17-state;
+		};
+		ufs@1d84000 {
+			/delete-property/ qcom,ice;
+		};
+		geniqup@9c0000 {
+			i2c@988000 {
+				/delete-property/ interconnects;
+			};
+		};
+		geniqup@8c0000 {
+			i2c@890000 {
+				/delete-property/ interconnects;
+			};
+		};
+		geniqup@ac0000 {
+			i2c@a90000 {
+				/delete-property/ interconnects;
+			};
+			serial@a8c000 {
+				/delete-property/ interconnects;
+			};
+			i2c@a94000 {
+				/delete-property/ interconnects;
+			};
+		};
+		usb@a6f8800 {
+			/delete-property/ interconnects;
+		};
+		usb@a8f8800 {
+			/delete-property/ interconnects;
+		};
+		usb@a4f8800 {
+			/delete-property/ interconnects;
+		};
+		timer@17c20000 {
+			/delete-node/ frame@17c23000;
+			/delete-node/ frame@17c25000;
+			/delete-node/ frame@17c27000;
+			/delete-node/ frame@17c29000;
+			/delete-node/ frame@17c2b000;
+			/delete-node/ frame@17c2d000;
+		};
+	};
+	reserved-memory {
+			/delete-node/ sail-ss@80000000;
+			/delete-node/ hyp@90000000;
+			/delete-node/ xbl-boot@90600000;
+			/delete-node/ aop-image@90800000;
+			/delete-node/ aop-cmd-db@90860000;
+			/delete-node/ uefi-log@908b0000;
+			/delete-node/ ddr-training-checksum@908c0000;
+			/delete-node/ reserved@908f0000;
+			/delete-node/ secdata-apss@908fe000;
+			/delete-node/ tz-sail-mailbox@90c00000;
+			/delete-node/ sail-ss@90d00000;
+			/delete-node/ sail-ss@90e00000;
+			/delete-node/ aoss-backup@91b00000;
+			/delete-node/ cpucp-backup@91b40000;
+			/delete-node/ tz-config-backup@91b80000;
+			/delete-node/ ddr-training-data@91b90000;
+			/delete-node/ cdt-data-backup@91ba0000;
+			/delete-node/ lpass-machine-learning@93b00000;
+			/delete-node/ adsp-rpc-remote-heap@94a00000;
+			/delete-node/ pil-camera@95200000;
+			/delete-node/ pil-adsp@95c00000;
+			/delete-node/ pil-gdsp0@97b00000;
+			/delete-node/ pil-gdsp1@99900000;
+			/delete-node/ pil-cdsp0@9b800000;
+			/delete-node/ pil-gpu@9d600000;
+			/delete-node/ pil-cdsp1@9d700000;
+			/delete-node/ pil-cvp@9f500000;
+			/delete-node/ pil-video@9fc00000;
+			/delete-node/ audio-mdf-region@ae000000;
+			/delete-node/ firmware-region@b0000000;
+			/delete-node/ hyptz-reserved@beb00000;
+			/delete-node/ scmi-region@d0000000;
+			/delete-node/ firmware-logs@d0040000;
+			/delete-node/ firmware-audio@d0050000;
+			/delete-node/ firmware-reserved@d0054000;
+			/delete-node/ firmware-quantum-test@d00f0000;
+			/delete-node/ tags@d0100000;
+			/delete-node/ qtee@d1300000;
+			/delete-node/ deepsleep-backup@d1800000;
+			/delete-node/ trusted-apps@d1900000;
+			/delete-node/ tz-stat@db100000;
+			/delete-node/ cpucp-fw@db200000;
+	};
+	smem {
+		/delete-property/ hwlocks;
+	};
+	/delete-node/ thermal-zones;
+	/delete-node/ dummy-sink;
+	/delete-node/ cpus;
+	/delete-node/ interconnect-aggre1-noc;
+	/delete-node/ interconnect-aggre2-noc;
+	/delete-node/ interconnect-clk-virt;
+	/delete-node/ interconnect-config-noc;
+	/delete-node/ interconnect-dc-noc;
+	/delete-node/ interconnect-gem-noc;
+	/delete-node/ interconnect-gpdsp-anoc;
+	/delete-node/ interconnect-lpass-ag-noc;
+	/delete-node/ interconnect-mc-virt;
+	/delete-node/ interconnect-mmss-noc;
+	/delete-node/ interconnect-nspa-noc;
+	/delete-node/ interconnect-nspb-noc;
+	/delete-node/ interconnect-pcie-anoc;
+	/delete-node/ interconnect-system-noc;
+	/delete-node/ pmu;
+};
 
+/* Deleting nodes using tags */
+/delete-node/ &pcie0;
+/delete-node/ &pcie0_ep;
+/delete-node/ &pcie0_phy;
+/delete-node/ &pcie1;
+/delete-node/ &pcie1_ep;
+/delete-node/ &pcie1_phy;
+/delete-node/ &ethernet1;
+/delete-node/ &ethernet0;
+/delete-node/ &stm;
+/delete-node/ &tmc_etf;
+/delete-node/ &adreno_smmu;
+/delete-node/ &pcie_smmu;
+/delete-node/ &gpucc;
+/delete-node/ &serdes0;
+/delete-node/ &serdes1;
+/delete-node/ &pdc;
+/delete-node/ &rng;
+/delete-node/ &ipcc;
+/delete-node/ &aoss_qmp;
+/delete-node/ &aoss_cti;
+/delete-node/ &tsens0;
+/delete-node/ &tsens1;
+/delete-node/ &tsens2;
+/delete-node/ &tsens3;
+/delete-node/ &ice;
+/delete-node/ &tcsr_mutex;
+/delete-node/ &cpufreq_hw;
+/delete-node/ &i2c0;
+/delete-node/ &spi0;
+/delete-node/ &i2c1;
+/delete-node/ &spi1;
+/delete-node/ &spi2;
+/delete-node/ &i2c3;
+/delete-node/ &spi3;
+/delete-node/ &i2c4;
+/delete-node/ &spi4;
+/delete-node/ &i2c5;
+/delete-node/ &spi5;
+/delete-node/ &uart5;
+/delete-node/ &qupv3_id_3;
+/delete-node/ &i2c7;
+/delete-node/ &spi7;
+/delete-node/ &i2c8;
+/delete-node/ &spi8;
+/delete-node/ &i2c9;
+/delete-node/ &spi9;
+/delete-node/ &uart9;
+/delete-node/ &i2c10;
+/delete-node/ &spi10;
+/delete-node/ &spi11;
+/delete-node/ &spi12;
+/delete-node/ &uart12;
+/delete-node/ &i2c13;
+/delete-node/ &i2c14;
+/delete-node/ &spi14;
+/delete-node/ &i2c15;
+/delete-node/ &spi15;
+/delete-node/ &i2c16;
+/delete-node/ &spi16;
+/delete-node/ &i2c17;
+/delete-node/ &spi17;
+/delete-node/ &uart17;
+/delete-node/ &spi18;
+/delete-node/ &i2c19;
+/delete-node/ &spi19;
+/delete-node/ &i2c20;
+/delete-node/ &spi20;
+/delete-node/ &sram;
diff --git a/configs/qcs9100_defconfig b/configs/qcs9100_defconfig
index 1201753dd08..d5a22b73600 100644
--- a/configs/qcs9100_defconfig
+++ b/configs/qcs9100_defconfig
@@ -22,6 +22,76 @@ CONFIG_SYS_LOAD_ADDR=0xD3500000
 CONFIG_CMD_BOOTQLI=y
 
 CONFIG_DM_PCA953X=y
+
+# CONFIG_BOOTP_BOOTPATH is not set
+# CONFIG_BOOTP_DNS is not set
+# CONFIG_BOOTP_GATEWAY is not set
+# CONFIG_BOOTP_HOSTNAME is not set
+# CONFIG_BOOTP_SUBNETMASK is not set
+# CONFIG_BOOTP_PXE is not set
+# CONFIG_NET_TFTP_VARS is not set
+# CONFIG_BOOTDEV_ETH is not set
+# CONFIG_CMD_EEPROM is not set
+# CONFIG_CLK_QCOM_APQ8016 is not set
+# CONFIG_CLK_QCOM_APQ8096 is not set
+# CONFIG_CLK_QCOM_QCM2290 is not set
+# CONFIG_CLK_QCOM_QCS404 is not set
+# CONFIG_CLK_QCOM_SDM845 is not set
+# CONFIG_CLK_QCOM_SM6115 is not set
+# CONFIG_CLK_QCOM_SM8150 is not set
+# CONFIG_CLK_QCOM_SM8250 is not set
+# CONFIG_CLK_QCOM_SM8550 is not set
+# CONFIG_CLK_QCOM_SM8650 is not set
+# CONFIG_CLK_QCOM_SC7280 is not set
+# CONFIG_DFU_MMC is not set
+# CONFIG_DM_KEYBOARD is not set
+# CONFIG_BUTTON_KEYBOARD is not set
+# CONFIG_NVMEM is not set
+# CONFIG_MMC_HS200_SUPPORT is not set
+# CONFIG_MMC_SDHCI is not set
+# CONFIG_MMC_SDHCI_ADMA is not set
+# CONFIG_MMC_SDHCI_MSM is not set
+# CONFIG_PHY_MICREL is not set
+# CONFIG_PHY_MICREL_KSZ90X1 is not set
+# CONFIG_DM_MDIO is not set
+# CONFIG_DM_ETH_PHY is not set
+# CONFIG_DWC_ETH_QOS is not set
+# CONFIG_DWC_ETH_QOS_QCOM is not set
+# CONFIG_PINCTRL_QCOM_APQ8016 is not set
+# CONFIG_PINCTRL_QCOM_APQ8096 is not set
+# CONFIG_PINCTRL_QCOM_QCM2290 is not set
+# CONFIG_PINCTRL_QCOM_QCS404 is not set
+# CONFIG_PINCTRL_QCOM_SDM845 is not set
+# CONFIG_PINCTRL_QCOM_SM6115 is not set
+# CONFIG_PINCTRL_QCOM_SM8150 is not set
+# CONFIG_PINCTRL_QCOM_SM8250 is not set
+# CONFIG_PINCTRL_QCOM_SM8550 is not set
+# CONFIG_PINCTRL_QCOM_SM8650 is not set
+# CONFIG_VIDEO is not set
+# CONFIG_VIDEO_FONT_16X32 is not set
+# CONFIG_SYS_WHITE_ON_BLACK is not set
+# CONFIG_NO_FB_CLEAR is not set
+# CONFIG_VIDEO_SIMPLE is not set
+# CONFIG_VIDEO_LOGO is not set
+# CONFIG_BACKLIGHT is not set
+# CONFIG_VIDEO_BPP8 is not set
+# CONFIG_VIDEO_BPP16 is not set
+# CONFIG_VIDEO_BPP32 is not set
+# CONFIG_CONSOLE_NORMAL is not set
+# CONFIG_PANEL is not set
+# CONFIG_SIMPLE_PANEL is not set
+# CONFIG_MMC is not set
+# CONFIG_MMC_WRITE is not set
+# CONFIG_DM_MMC is not set
+# CONFIG_MMC_QUIRKS is not set
+# CONFIG_MMC_HW_PARTITIONING is not set
+# CONFIG_MMC_VERBOSE is not set
+# CONFIG_MMC_SDHCI_ADMA_64BIT is not set
+# CONFIG_ETH is not set
+# CONFIG_DM_ETH is not set
+# CONFIG_NETDEVICES is not set
+# CONFIG_CMD_NET is not set
+
 CONFIG_DEFAULT_DEVICE_TREE="qcom/qcs9100-ride-r3"
 # CONFIG_ENV_IS_DEFAULT is not set
 # CONFIG_ENV_IS_NOWHERE is not set
-- 
2.34.1

